File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode3375MinimumOperationsToMakeArrayValuesEqualToK.cpp
#include "LeetCode3375MinimumOperationsToMakeArrayValuesEqualToK.h"

#include <algorithm>
#include <unordered_set>
using namespace std;
int LeetCode3375MinimumOperationsToMakeArrayValuesEqualToK::minOperations(vector<int>& nums, int k)
{
    unordered_set<int> set{};
    int min=numeric_limits<int>::max();
    for (auto num:nums)
    {
        set.insert(num);
        min = std::min(num, min);
    }
    if (min<k)
        return -1;
    const int count=static_cast<int>( set.size());
    return min==k?count-1:count;
}

File: LeetCodeProblemsGo/LeetCode3375MinimumOperationsToMakeArrayValuesEqualToK/LeetCode3375MinimumOperationsToMakeArrayValuesEqualToK.go
package LeetCode3375MinimumOperationsToMakeArrayValuesEqualToK

import "math"

func minOperations(nums []int, k int) int {
	numsMap := make(map[int]struct{})
	min := math.MaxInt32
	for _, num := range nums {
		numsMap[num] = struct{}{}
		if num < min {
			min = num
		}
	}
	if min < k {
		return -1
	}
	distinctCount := len(numsMap)
	if min == k {
		return distinctCount - 1
	}
	return distinctCount

}
